lib/rubocop/cop/lint/number_conversion.rb

```diff
@@ -49,6 +59,11 @@ def on_send(node)
 
         private
 
+        def date_time_object?(node)
+          time?(node) || date_time?(node) ||
+            time?(node.children[0]) || date_time?(node.children[0])
```

Comment:
@rrosenblum Is there a way to define matcher that will check this recursively for children? Currently this works for `Time.now.to_i` and `Time.zone.now.to_i` but will fail for `Time.zone.now.to_datetime.to_i`.

Reply:
This is always a bit tricky to handle. You can define a node matcher that can check a node's ancestors' pattern, but you still have to tell it exactly how to match so it doesn't work for checking both parent and grandparent. 

I added a method in `Style/SafeNavigationChain` to return the full method chain. This method should probably be moved to `SendNode` or somewhere similar so that it is more accessible.

```ruby
        def method_chain(node)    
          chain = node    
          while chain.send_type?    
            chain = chain.parent if chain.parent &&    
                                    %i[send csend].include?(chain.parent.type)    
            break                                                                                                                                  
          end           
          chain                     
        end
```

You can combine this with a `def_node_matcher` or `def_node_search` that can then check if the start of the chain matches your pattern. I normally wind up using `def_node_matcher` because it can return match groups. 

Let me know if you need anymore help with this.

Reply:
Thanks @rrosenblum! That helped.